tests/django_pg_migration_tools/test_operations.py

```diff
@@ -924,6 +924,153 @@ def test_when_not_unique_constraint(self):
+        # later.
+        assert queries[1]["sql"] == "SHOW lock_timeout;"
+        # 3. Remove the timeout.
+        assert queries[2]["sql"] == "SET lock_timeout = '0';"
+        # 4. Verify if the index is invalid.
+        assert queries[3]["sql"] == dedent(
+            """
+            SELECT relname
+            FROM pg_class, pg_index
+            WHERE (
+                pg_index.indisvalid = false
+                AND pg_index.indexrelid = pg_class.oid
+                AND relname = 'unique_int_field'
+            );
+            """
+        )
+        # 5. Finally create the index concurrently.
+        assert (
+            queries[4]["sql"]
+            == 'CREATE UNIQUE INDEX CONCURRENTLY IF NOT EXISTS "unique_int_field" ON "example_app_intmodel" ("int_field") WHERE "int_field" >= 2'
+        )
+        # 6. Set the timeout back to what it was originally.
+        assert queries[5]["sql"] == "SET lock_timeout = '1s';"
+
+        assert len(queries) == 6
+        # there is not constraint
+
+        # Reverse the migration to drop the index and constraint, and verify
+        # that the lock_timeout queries are correct.
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as reverse_queries:
+                operation.database_backwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+        # 1. Check that the constraint is still there.
+        assert reverse_queries[0]["sql"] == dedent(
+            """
+            SELECT conname
+            FROM pg_catalog.pg_constraint
+            WHERE conname = 'unique_int_field';
+            """
+        )
+
+        # 2. perform the ALTER TABLE.
+        assert reverse_queries[1]["sql"] == "SHOW lock_timeout;"
+
+        # 3. Remove the timeout.
+        assert reverse_queries[2]["sql"] == "SET lock_timeout = '0';"
+        # 4. Verify if the index is invalid.
+        assert (
+            reverse_queries[3]["sql"]
+            == 'DROP INDEX CONCURRENTLY IF EXISTS "unique_int_field"'
+        )
+
+        assert reverse_queries[4]["sql"] == "SET lock_timeout = '1s';"
+
+        assert len(reverse_queries) == 5
+
+        # Verify the constraint doesn't exist any more.
+        with connection.cursor() as cursor:
+            cursor.execute(
+                _CHECK_INDEX_EXISTS_QUERY,
+                {
+                    "table_name": "example_app_intmodel",
+                    "index_name": "unique_int_field",
+                },
+            )
+            assert not cursor.fetchone()
+
+            cursor.execute(
+                _CHECK_CONSTRAINT_EXISTS_QUERY,
+                {
+                    "table_name": "example_app_intmodel",
+                    "constraint_name": "unique_int_field",
+                },
+            )
+            assert cursor.fetchone() is None
+
```

Comment:
Suggestion: [✂️] there was never a constraint here, only an index. So probably better to remove this assertion.